src/sentry/event_manager.py

```diff
@@ -1337,6 +1331,44 @@ def get_culprit(data: Mapping[str, Any]) -> str:
     )
 
 
+def assign_event_to_group(event: Event, job: Job, metric_tags: MutableTags) -> Optional[GroupInfo]:
```

Comment:
I believe this would be more appropriate return type
```suggestion
def assign_event_to_group(event: Event, job: Job, metric_tags: MutableTags) -> GroupInfo | None:
```

Reply:
You can follow this up on the next PR if you want to merge it now

Reply:
Oh, yeah, I guess we can do that now, without even using `__future__`, huh? Yay for finally being off of Python 3.8!

Reply:
Finally 🚀